bibo/internals.py

```diff
@@ -192,3 +195,50 @@ def complete_path(ctx, args, incomplete):
         if os.access(path, os.R_OK):
             options.append((path, os.path.basename(path)))
     return options
+
+
+def match_case(substring, target):
+    """
+    Return the string `substring` with the same case as in the `target`
+    string.
+    """
+    index = target.lower().find(substring.lower())
+    if index < 0:
```

Comment:
Eh, I'm not sure how I feel about `match_case` raising. What if there are multiple matches here? For example, this might be used on abstract fields right? If this is going to just find a single hit, which we know already exists (implied based on usage), maybe this should be renamed - `substring_orig_case` or something.

Reply:
You raised some good points here! I'm happy with renaming the function. ATM it matches only the first occurrence. We can improve this to always return a list with all matches, and then there's also no reason to raise. What do you think?

Reply:
I feel like returning a list (al matches), and just using something like `re.findall`, might be better here. I feel like that's what we want to happen?

Reply:
Yep! Improving this now.